### DISCUSSION TITLE
Sending messages as a public channel

### DISCUSSION THREAD

**Discussion by foodornt** [gram-js/gramjs#393] [Q&A] [answered]
Using telegram you can choose which identity you want to send your message with

how can I do that with gramjs?

**Comment by rojvv** [comment, marked as answer]
Seems like the helper methods currently don't support this parameter, but you can always use raw methods, like this:

```ts
await client.invoke(
  new Api.messages.SendMessage({
    peer: "chat to send the message to",
    message: "message text",
    sendAs: "chat to send the message as",
  }),
);
```

**Reply by foodornt** [reply]
Absolute!
One more thing: how can I mention users this way? the `[mention](userid)` way isn't working

_Both MarkdownParser and HTMLParser don't seem to recognize mentions_

**Reply by rojvv** [reply]
You can pass entities with the `entities` parameter.
